src/main/java/org/traccar/protocol/TopinProtocolDecoder.java

```diff
@@ -122,13 +150,7 @@ protected Object decode(
             position.set(Position.KEY_VERSION_FW, firmware);
 
             ByteBuf content = Unpooled.buffer();
-            content.writeByte(battery);
```

Comment:
Why are we removing all this?

Reply:
As stated in the documentation, the TOPIN protocol expects only the heartbeat interval as an answer to the 0x13 status message and not all other parameters.

**0x13 status package**
The device is sent to the server
Start bit 2byte packet length 1byte protocol number 1byte battery power 1byte software version
1byte time zone 1byte status packet upload interval 1byte end bit 2byte
Eg.7878 06 13 55 23 08 03 0D0A
55 means that the battery power is 85%, 23 said the software version number 35 08 that is the
East 8 area (only integer), 03 for the state packet upload interval of 3 minutes
Start bit 2byte packet length 1byte protocol number 1byte status packet upload interval 1byte
end bit 2byte
This data is sent to the server for the server, used to set the status packet upload interval
**Eg.7878 02 13 03 0D0A**
03 to set the heartbeat packet upload interval of 3 minutes to upload once, in minutes

Reply:
Have you checked all other documents?

Reply:
As there is only one document for the specific TOPIN protocol and the decoder in traccar is specific for this given protocol and it matches with the TOPIN default server behaviour, there should not be an issue!